src/auth/redisClient.ts

```diff
@@ -0,0 +1,80 @@
+const redis = require("redis");
+const crypto = require("crypto");
+const bluebird = require("bluebird");
+const cryptojs = require("crypto-js");
+
+const HOUR_IN_SECONDS = 3600;
+
+// Adding promises to redis
+// https://stackoverflow.com/a/54844935
+
+// https://cloud.google.com/community/tutorials/nodejs-redis-on-appengine
+const client = (() => {
+  bluebird.promisifyAll(redis);
+  if (process.env.REDIS_PORT) {
+    const authParams: {
+      [index: string]: any;
+    } = {};
+    if (process.env.REDIS_KEY) {
+      authParams.auth_pass = process.env.REDIS_KEY;
+      authParams.return_buffers = true;
+    }
+    return redis.createClient(
+      process.env.REDIS_PORT,
+      process.env.REDIS_HOST,
+      authParams
+    );
+  } else {
+    return redis.createClient();
+  }
+})();
+
+const hash = (text: any) =>
+  crypto.createHash("sha256").update(text).digest("hex");
+const defaultKeyWord = "12345";
+// https://github.com/brix/crypto-js#plain-text-encryption
+const encryptAccessToken = (accessToken: any) =>
+  cryptojs.AES.encrypt(
+    accessToken,
+    process.env.EBAY_ACCESS_TOKEN_SECRET ?? defaultKeyWord
+  ).toString();
+
+const decryptAccessToken = (encryptedToken: any) =>
+  cryptojs.AES.decrypt(
+    encryptedToken,
+    process.env.EBAY_ACCESS_TOKEN_SECRET ?? defaultKeyWord
+  ).toString(cryptojs.enc.Utf8);
+
+export const setAccessToken = async (refreshToken: any, accessToken: any) => {
+  if (!client) {
+    return;
+  }
+  const hashedRefreshToken = hash(refreshToken);
+  const encryptedAccessToken = encryptAccessToken(accessToken);
+  // Auto remove it after an hour -> https://redis.io/commands/set
+  await client.setAsync(
+    hashedRefreshToken,
+    encryptedAccessToken,
+    "EX",
+    2 * (HOUR_IN_SECONDS - 120) //set expiry to be after 1 hour 58 minutes
```

Comment:
I think this ends up being 1 hour 56 minutes, which is fine, but might want to update it.
2 * (3600 - 120) = 7200 - 240 = 6960 => 116 minutes